main.py

```diff
@@ -111,16 +342,47 @@ def disable_pipeline(self, request: Request) -> JSONResponse:
         """Disable pipeline"""
         pipeline_id = request.path_params["pipeline_id"]
         log.debug(f"disable_pipeline /v1/pipeline/{pipeline_id}/disable")
-        pipeline = self.pipeline_controller.update_status(pipeline_id,
-                                                          "disabled")
-        if not pipeline:
-            msg = f"Pipeline {pipeline_id} not found"
-            log.debug(f"disable_pipeline result {msg} 404")
-            raise HTTPException(404, detail=msg)
+        pipeline = self.pipeline_controller.get_active_pipeline()
+        status = pipeline.get("status")
+        id_ = pipeline.get("id")
+        if pipeline and (id_ != pipeline_id or status == "enabling"):
+            msg = f"Other pipeline {id_} is {status}"
+            log.debug(f"disable_pipeline result {msg} 409")
+            raise HTTPException(409, detail=msg)
+
+        if status != "disabled":
+            pipeline = self.pipeline_controller.disabling_pipeline(pipeline_id)
+            if not pipeline:
+                msg = f"Pipeline {pipeline_id} not found"
+                log.debug(f"disable_pipeline result {msg} 404")
+                raise HTTPException(404, detail=msg)
+            self.load_pipeline(self.default_pipeline)
         msg = f"Pipeline {pipeline_id} disabled"
         log.debug(f"disable_pipeline result {msg} 200")
         return JSONResponse(msg)
 
+    @listen_to("kytos/flow_manager.flow.added")
+    def on_flow_mod_added(self, event):
+        """Looking for recently added flows"""
+        self.handle_flow_mod_added(event)
+
+    def handle_flow_mod_added(self, event):
+        """Handle recently added flows"""
+
+    @listen_to("kytos/flow_manager.flow.error")
+    def on_flow_mod_error(self, event):
+        """Handle flow mod errors"""
+        self.handle_flow_mod_error(event)
```

Comment:
`"kytos/flow_manager.flow.error"` isn't still being handled, if a user install a table-miss flow poorly formatted or with unsupported action the pipeline might not end up fully functional. At this point, since we're no longer rolling back should it should update the status of the pipeline as `enabling_error`? Let me know if you had other things in mind, it's important to simulate an actual OFPT_ERROR and see how it'll behave, one day a network engineer will exercise this path here, either accidentally or not :)

Reply:
Checked with:
```
{
    "flows": [
        {
            "cookie": 12465963768561532929,
            "priority": 100,
            "match": {
                "in_port": 1
            },
            "actions": [
                {
                    "action_type": "push_int"
                }
            ]
        }
    ]
}
```

Reply:
Cool. Let's close this thread.